signac/contrib/job.py

```diff
@@ -24,31 +23,170 @@
 logger = logging.getLogger(__name__)
 
 
-class _sp_save_hook:
-    """Hook to handle job migration when state points are changed.
-
-    When a job's state point is changed, in addition
-    to the contents of the file being modified this hook
-    calls :meth:`~Job._reset_sp` to rehash the state
-    point, compute a new job id, and move the folder.
-
-    Parameters
-    ----------
-    jobs : iterable of `Jobs`
-        List of jobs(instance of `Job`).
+class _StatepointDict(JSONDict):
```

Comment:
Sooooooo. Statepoint vs. state point. Probably needs to be:
```suggestion
class _StatePointDict(JSONDict):
```